src/CFamily/CMake/CMakeSettingsProvider.cs

```diff
@@ -34,7 +34,23 @@ internal interface ICMakeSettingsProvider
         /// Attempts to find and parse a CMakeSettings.json under the given root directory.
         /// </summary>
         /// <returns>Returns null if the file was not found or could not be parsed.</returns>
-        CMakeSettings TryGet(string rootDirectory);
+        CMakeSettingsSearchResult Find(string rootDirectory);
+    }
+
+    internal class CMakeSettingsSearchResult
+    {
+        public CMakeSettingsSearchResult(CMakeSettings parsedSettings, 
+            string cMakeSettingsFilePath, 
+            string rootCMakeListsFilePath)
+        {
+            ParsedSettings = parsedSettings;
+            CMakeSettingsFilePath = cMakeSettingsFilePath;
+            RootCMakeListsFilePath = rootCMakeListsFilePath;
+        }
+
+        public CMakeSettings ParsedSettings { get; }
```

Comment:
Minor: I don't think having `Parsed` in the property name adds any useful information - it could just be `Settings` / `AllSettings`.